rfcs/detecting_sensitive_textual_content/20230309-project_proposal_detecting_sensitive_textual_content.md

```diff
@@ -0,0 +1,379 @@
+  transformation to potentially several images on a single page
+- Extremely minor and only barely relevant: If we calculated the BlurHash for
+  all images in the future in our re-crawler, we could use them as placeholders
+  rather than the skeleton boxes we currently use
+
+#### Recommendation
+
+If we can spare the time, I think BlurHash is the correct choice for the
+following reasons:
+
+1. While significantly more complex, it is not sufficiently complex to be
+   burdensome. The simplicity of the CSS filter makes BlurHash look monstrous,
+   but in the grand scheme of things, it's actually a relatively simple approach
+   to blurring the images.
+2. It avoids performance issues with relying on low-spec clients to apply
+   Gaussian blurs to potentially several images.
+3. BlurHash hashes have significant future value and all potential API level
+   performance issues can be ameliorated through caching hashes with Redis.
+
+The only significant downside that I think exists is the delay between
+unblurring and displaying the full image due to needing to download the
+full unblurred image. However, I think this is balanced out by the bytes saved
+over the network in _not_ sending images that might potentially never be
+unblurred.
+
+For reviewers: **What are you feelings on this issue and why?**. One alternative
+to going straight for BlurHash, if we wanted to get blurring out the door
+quickly is to treat BlurHash as a future "nice to have" and use CSS blurring in
+the meantime. This would allow us to get a fully working feature slightly more
+quickly. I'm not totally convinced it'd be _that_ much faster though. Like I
+said above, BlurHash looks complicated, but really only in comparison to how
+simple the CSS approach is. Anything would look complicated when compared to
+just adding a CSS filter style to something.
+
+##### Important detail
+
+The success of this project rests on the completion of its technical
+implementation. However, we can add the following events to the API and frontend
+to better understand the usage of the features.
+
+### API
+
+- `SENSITIVE_RESULT_COUNT`
+  - Sent for each query where the search param `include_sensitive_results` is
+    set to `True`. It should include a raw count of the number of results with
+    sensitive textual content or with `mature` set to `True`. Not sent for
+    results where the search params would not include sensitive results.
+
+### Frontend
+
+- `SENSITIVE_RESULTS_TOGGLED`
+  - Sent each time the "include sensitive results" setting is toggled. Include a
+    prop noting if the setting was being toggled on or off.
+- `SENSITIVE_RESULT_UNBLURRED`
+  - Sent each time a result is unblurred. This should include a property
+    denoting whether it on the single result or search results page.
+- `SENSITIVE_RESULT_REBLURRED`
+  - Sent each time a result is reblurred. This should include a property
+    denoting whether it on the single result or search results page.
```

Comment:
Would we include the media ID with these?

Reply:
I can't think of how we would use it now but considering it is free to include props on custom events, I don't see why not.